src/NATS.Client.KeyValueStore/NatsKVStore.cs

```diff
@@ -44,13 +44,23 @@ public class NatsKVStore : INatsKVStore
     private const string NatsSequence = "Nats-Sequence";
     private const string NatsTimeStamp = "Nats-Time-Stamp";
     private static readonly Regex ValidKeyRegex = new(pattern: @"\A[-/_=\.a-zA-Z0-9]+\z", RegexOptions.Compiled);
```

Comment:
Side note, have we looked into whether it's worth doing sourcegen regex here for NET8+? If not I can file issue to investigate further.

Reply:
i didn't think of that actually we should please do

Reply:
You can do source generated as well. 

```
[GeneratedRegex(@"\A[-/_=.a-zA-Z0-9]+\z", RegexOptions.Compiled)]
private static partial Regex ValidKeyRegex();
```